clis/teliod/src/system_configurator.rs

```diff
@@ -0,0 +1,114 @@
+    /// Configure the IP address and routes for a given interface
+    fn set_ip(&self, adapter_name: &str, ip_address: &IpAddr) -> Result<(), TeliodError>;
+
+    /// Removes an IP address from a given interface
+    fn remove_ip(&self, adapter_name: &str) -> Result<(), TeliodError>;
+
+    /// Helper function to execute a system command
+    fn execute(&self, command: &mut Command) -> Result<(), TeliodError> {
+        let output = command
+            .output()
+            .map_err(|e| TeliodError::SystemCommandFailed(e.to_string()))?;
+
+        if output.status.success() {
+            Ok(())
+        } else {
+            let stderr = String::from_utf8_lossy(&output.stderr);
+            error!("Error executing command {:?}: {:?}", command, stderr);
+            Err(TeliodError::SystemCommandFailed(stderr.into()))
+        }
+    }
+}
+
+/// Implementation using `ifconfig`
+pub struct IfconfigSystemConfigurator;
+
+impl SystemConfigurator for IfconfigSystemConfigurator {
+    fn set_ip(&self, adapter_name: &str, ip_address: &IpAddr) -> Result<(), TeliodError> {
+        let ip_string = ip_address.to_string();
+        let cidr_suffix = if ip_address.is_ipv4() { "/10" } else { "/64" };
+        let cidr_string = format!("{}{}", ip_string, cidr_suffix);
+        let ip_type = if ip_address.is_ipv4() {
+            "inet"
+        } else {
+            "inet6"
+        };
+
+        info!(
+            "Assigning IP address for {} to {}",
+            adapter_name, cidr_string
+        );
+
+        match std::env::consts::OS {
+            "macos" => {
+                self.execute(Command::new("ifconfig").args([
+                    adapter_name,
+                    ip_type,
+                    &cidr_string,
+                    &ip_string,
+                ]))?;
+
+                if ip_address.is_ipv4() {
+                    self.execute(Command::new("route").args(["add", "100.64/10", &ip_string]))?;
+                } else {
+                    self.execute(Command::new("route").args([
+                        "add",
+                        "-inet6",
+                        "fd74:656c:696f::/64",
+                        &ip_string,
+                    ]))?;
+                }
+            }
+            _ => {
+                self.execute(Command::new("ifconfig").args([adapter_name, ip_type, &cidr_string]))?;
+            }
+        }
+
+        self.execute(Command::new("ifconfig").args([adapter_name, "mtu", "1420"]))?;
+        self.execute(Command::new("ifconfig").args([adapter_name, "up"]))
+    }
+
+    fn remove_ip(&self, adapter_name: &str) -> Result<(), TeliodError> {
+        info!("Removing assigned IP from {}", adapter_name);
+
+        self.execute(Command::new("ifconfig").args([adapter_name, "delete"]))?;
+        self.execute(Command::new("ifconfig").args([adapter_name, "down"]))
+    }
+}
+
+/// Mock implementation of `SystemConfigurator` for testing purposes.
+pub struct MockSystemConfigurator {
```

Comment:
This should be part of an eventual tests mod 🤔 

Also, you've defined a mock for the `SystemConfigurator` but looks like the tests are missing.

Reply:
The tests are in `[nat-lab/tests/test_teliod.py](https://github.com/NordSecurity/libtelio/pull/953/files/950ef94273060333616143318a543dffbd6878af#diff-ad0957b95a003082412bb199243caf3ab119b9af35cd68110a91332239101c73)`

Reply:
As mentioned [here](https://github.com/NordSecurity/libtelio/pull/953#discussion_r1853636572), tests from [nat-lab/tests/test_teliod.py](https://github.com/NordSecurity/libtelio/pull/953/files/950ef94273060333616143318a543dffbd6878af#diff-ad0957b95a003082412bb199243caf3ab119b9af35cd68110a91332239101c73) can be made unit tests. We shouldn't be modifying production code for integration testing.

Reply:
Removed the "Mock" implementation in favor of "Manual" interface config